File: 백준/Bronze/13300. 방 배정/방 배정.cc
#include <iostream>
using namespace std;

int main(){
    int n, k, s, y, count = 0;
    cin >> n >> k;
    
    int rooms[6][2] = {0}; // 학년별(0~5), 성별(0:남자, 1:여자)
    
    for (int i = 0; i < n; i++){
        cin >> s >> y;    // 성별, 학년
        rooms[y-1][s]++;  // 학년은 1부터 시작하므로 y - 1
    }
    
    for (int i = 0; i < 6; i++){        // 학년 반복
        for (int j = 0; j < 2; j++){    // 성별 반복
            count += (rooms[i][j] + k - 1) / k; // 올림 계산
        }
    }
    
    cout << count << endl;
    return 0;
}

File: 백준/Bronze/13300. 방 배정/방 배정.py
n, k = map(int, input().split())
rooms = [[0] * 2 for _ in range(6)]  # 학년별, 성별 학생 수를 저장 (6학년, 성별 2개)

for _ in range(n):
    s, y = map(int, input().split())  # 성별(s), 학년(y)
    rooms[y - 1][s] += 1

count = 0
for grade in rooms:
    for students in grade:
        count += (students + k - 1) // k  # 학생 수를 k로 나눈 방의 개수 계산

print(count)
